polyply/src/gen_coords.py

```diff
@@ -74,8 +74,9 @@ def _initialize_cylces(topology, cycles, tolerance):
         for mol_idx in topology.mol_idx_by_name[mol_name]:
             # initalize as dfs tree
             molecule = topology.molecules[mol_idx]
-            if set(dict(nx.degree(molecule)).values()) != {1, 2}:
-                raise IOError("Only linear cyclic molecules are allowed at the moment.")
+            cycles = nx.cycle_basis(molecule)
```

Comment:
Probably fine here, but be aware there are (many) issues with using the cycle_basis: https://depth-first.com/articles/2020/08/31/a-smallest-set-of-smallest-rings/

Reply:
That's fine. In the end this is only a first line of defense. I believe if it is topologically not allowed the constraint algorithm will most likekly raise an error.